### PR TITLE
Set up a basic multi sig workflow with signers

### PR DESCRIPTION
**Pull request by Carl-Foster**

### CODE REVIEW

File: src/components/SignTransactionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+import { HorizontalLabelledField } from './LabelledField'
+
+interface State {
+  transaction?: Transaction
+  signed: boolean
+}
+export class SignTransactionForm extends React.Component<SignTransactionFormProps, State> {
+  state: State = {
+    signed: false,
+  }
```
Comment (desnor) [inline src/components/SignTransactionForm.tsx]: do we want to have an initial state value for the `state.transaction`?

File: src/components/Transactions.tsx [hunk 2]
```diff
@@ -25,18 +25,16 @@ export class Transactions extends React.Component<Props> {
...
   render() {
-    const {
-      isLoading,
-      operations,
-    } = this.props
+    const { isLoading, operations } = this.props
     return (
       <React.Fragment>
-        <h1 className='sub-heading primary-font'>
+        <h1 className="sub-heading primary-font">
```
Comment (desnor) [inline src/components/Transactions.tsx]: shouldn't the linter have kept these as `'`? seems weird to revert to `"`

File: src/store/reducers/transfer.ts [hunk 3]
```diff
@@ -25,13 +26,16 @@ export const transfer = combineReducers<TransferState, RootAction>({
...
   isTransferring: (state = false, action) => {
     switch (action.type) {
-      case getType(transferRequest): return true
+      case getType(transferRequest):
+      case getType(transactionRequest):
+        return true
 
-      case getType(transferSuccess):
-      case getType(transferFailed):
+      case getType(transactionSuccess):
```
Comment (desnor) [inline src/store/reducers/transfer.ts]: Are we still handling the transfer success and failure somewhere?

File: src/components/TransactionOperation.tsx [hunk 4]
```diff
@@ -23,11 +23,11 @@ const baseOperationRecordKeys = [
...
 ]
 
-export const renderOperationRecords = (operation: OperationRecord) => {
+export const renderOperationRecords = (operation: OperationRecord | TransactionOperation) => {
   const keys = Object.keys(operation)
-    .filter((key) => typeof operation[key] === 'string')
-    .filter((key) => baseOperationRecordKeys.indexOf(key) === -1)
-  return keys.map((key) => (
+    .filter(key => typeof operation[key] === 'string')
+    .filter(key => baseOperationRecordKeys.indexOf(key) === -1)
```
Comment (desnor) [inline src/components/TransactionOperation.tsx]: can we use `includes` instead?

File: src/components/SignTransactionForm.tsx [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        {this.renderSigners()}
+      </div>
+    )
+  }
+}
+
+class OperationOverview extends React.Component<{ operation: TransactionOperation }> {
+  renderOperationRecords = () =>
+    Object.entries(this.props.operation)
+      .filter(([_, value]) => typeof value === 'string' || typeof value === 'number')
```
Comment (desnor) [inline src/components/SignTransactionForm.tsx]: these filter calls seem like they might read better the other way, what I mean is:

```js
.filter(([_, value]) => ['string','number'].includes(typeof value)) // wrap test values in array
.filter(([key]) => key !== 'type') // unwrap test key array
```

in which case we could do it at the same time without too much bloat?

```js
.filter(([key, value]) => key !== 'type && ['string','number'].includes(typeof value))
```

File: src/store/reducers/transfer.ts (hunk 3 above)
Comment (Carl-Foster) [reply src/store/reducers/transfer.ts]: I've made the transferRequest go through to the transaction caller to keep it DRY

File: src/components/SignTransactionForm.tsx (hunk 5 above)
Comment (Carl-Foster) [reply src/components/SignTransactionForm.tsx]: I like it

**Pull request merged**